deps/rabbit/test/backing_queue_SUITE.erl

```diff
@@ -1305,20 +1311,35 @@ variable_queue_ack_limiting2(VQ0, _Config) ->
     %% fetch half the messages
     {VQ4, _AckTags} = variable_queue_fetch(Len div 2, false, false, Len, VQ3),
 
-    VQ5 = check_variable_queue_status(
-            VQ4, [{len,                         Len div 2},
-                  {messages_unacknowledged_ram, Len div 2},
-                  {messages_ready_ram,          Len div 2},
-                  {messages_ram,                Len}]),
+    Status = variable_queue_status(VQ4),
+    VQ5 = VQ4,
+%    VQ5 = case proplists:get_value(mode, Status) of
+%        default ->
+%            check_variable_queue_status(
+%                VQ4, [{len,                         Len div 2},
+%                      {messages_unacknowledged_ram, Len div 2},
+%                      {messages_ready_ram,          Len div 2},
+%                      {messages_ram,                Len}]);
+%        lazy ->
+%            VQ4
+%            %% @todo This check has been broken for a very long time. (Or never worked.)
+%%            check_variable_queue_status(
+%%                VQ4, [{len,                         Len div 2},
+%%                      {messages_unacknowledged_ram, 0},
+%%                      {messages_ready_ram,          0},
+%%                      {messages_ram,                0}])
+%    end,
```

Comment:
Probably worth adding the check back if it is deterministic, of course independent of mode.